homeassistant/auth/providers/command_line.py

```diff
@@ -0,0 +1,160 @@
+"""Auth provider that validates credentials via an external command."""
+
+import typing as T
+
+import asyncio.subprocess
+import collections
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+
+from . import AuthProvider, AUTH_PROVIDER_SCHEMA, AUTH_PROVIDERS, LoginFlow
+from ..models import Credentials, UserMeta
+
+
+CONF_COMMAND = "command"
+CONF_ARGS = "args"
+CONF_META = "meta"
+
+CONFIG_SCHEMA = AUTH_PROVIDER_SCHEMA.extend({
+    vol.Required(CONF_COMMAND): vol.All(
+        str,
+        os.path.normpath,
+        msg="must be an absolute path"
+    ),
+    vol.Optional(CONF_ARGS, default=None): vol.Any(vol.DefaultTo(list), [str]),
+    vol.Optional(CONF_META, default=False): bool,
+}, extra=vol.PREVENT_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InvalidAuthError(HomeAssistantError):
+    """Raised when authentication with given credentials fails."""
+
+
+@AUTH_PROVIDERS.register("command_line")
+class CommandLineAuthProvider(AuthProvider):
+    """Auth provider validating credentials by calling a command."""
+
+    DEFAULT_TITLE = "Command Line Authentication"
+
+    # which keys to accept from a program's stdout
+    ALLOWED_META_KEYS = ("name",)
+
+    def __init__(self, *args: T.Any, **kwargs: T.Any) -> None:
+        """Extend parent's __init__.
+
+        Adds self._user_meta dictionary to hold the user-specific
+        attributes provided by external programs.
+        """
+        super().__init__(*args, **kwargs)
+        self._user_meta = {}  # type: T.Dict[str, T.Dict[str, T.Any]]
+
+    async def async_login_flow(self, context: T.Optional[T.Dict]) -> LoginFlow:
+        """Return a flow to login."""
+        return CommandLineLoginFlow(self)
+
+    async def async_validate_login(self, username: str, password: str) -> None:
+        """Validate a username and password."""
+        env = {
+            "username": username,
+            "password": password,
+        }
+        try:
+            # pylint: disable=no-member
+            process = await asyncio.subprocess.create_subprocess_exec(
```

Comment:
`asyncio.subprocess` already imported

Reply:
pylint mixes ``asyncio.subprocess`` with ``subprocess`` and complains about missing ``create_subprocess_exec`` therein, don't know why. I imported the right module.